Database/MySQL.php

```diff
@@ -11,67 +11,123 @@
     private $fileName;
     private $ignoreTables = '';
+    private $params;
 
     /**
      * DB Auth.
      *
-     * @param array  $params
+     * @param array $params
      * @param string $basePath
      */
     public function __construct($params, $basePath)
     {
         parent::__construct($basePath);
+        $this->params = $params['mysql'];
+    }
 
-        $params = $params['mysql'];
-        $this->allDatabases = $params['all_databases'];
-        $this->database     = $params['database'];
-        $this->auth         = '';
-
-        if ($this->allDatabases) {
+    /**
+     * Prepare a database name and a file dump name for mysqldump command
+     */
+    protected function prepareFileName()
+    {
+        if ($this->params['all_databases']) {
             $this->database = '--all-databases';
             $this->fileName = 'all-databases.sql';
         } else {
-            $this->fileName = $this->database.'.sql';
+            $this->database = $this->params['database'];
+            $this->fileName = $this->database . '.sql';
         }
+    }
 
-        if (isset($params['ignore_tables'])) {
-            foreach ($params['ignore_tables'] as $ignoreTable) {
-                if ($this->allDatabases) {
+    /**
+     * Prepare ignore tables attribute for mysqldump command
+     */
+    protected function prepareIgnoreTables()
+    {
+        if (isset($this->params['ignore_tables'])) {
+            foreach ($this->params['ignore_tables'] as $ignoreTable) {
+                if ($this->params['all_databases']) {
                     if (false === strpos($ignoreTable, '.')) {
                         throw new \LogicException(
                             'When dumping all databases both database and table must be specified when ignoring table'
                         );
                     }
                     $this->ignoreTables .= sprintf('--ignore-table=%s ', $ignoreTable);
                 } else {
-                    $this->ignoreTables .= sprintf('--ignore-table=%s.%s ', $this->database, $ignoreTable);
+                    $this->ignoreTables .= sprintf('--ignore-table=%s.%s ', $this->params['database'], $ignoreTable);
                 }
             }
         }
+    }
+
+    /**
+     * Prepare mysql configuration file for connection
+     */
+    protected function prepareConfigurationFile()
+    {
+        if ($this->params['db_user']) {
+            $cnfParams['user']  = $this->params['db_user'];
+            $cnfFile            = "[client]\n";
 
-        /* if user is set, we add authentification */
-        if ($params['db_user']) {
-            $this->auth = sprintf('-u%s', $params['db_user']);
+            if ($this->params['db_password']) {
+                $cnfParams += array(
+                    "password"  => $this->params['db_password'],
+                    "host"      => $this->params['db_host'],
+                    "port"      => $this->params['db_port']
+                );
+            }
```

Comment:
Ah =) okey.

No, I do not insist. I'll want your opinion. I dislike array_merge and the plus operator in this case because it could be avoided. 

How about this: 

```php
 if ($this->params['db_user']) {
  $cnfFile  = "[client]\n"; 
  $cnfParams['user']  = $this->params['db_user'];

  if ($this->params['db_password']) {
    $cnfParams['password']  = $this->params['db_password'];
    $cnfParams['host']  = $this->params['db_host'];
    $cnfParams['uporter']  = $this->params['db_port'];
  }
}

```